sklearn/utils/_tags.py

```diff
@@ -0,0 +1,78 @@
+import numpy as np
+
+_DEFAULT_TAGS = {
+    'non_deterministic': False,
+    'requires_positive_X': False,
+    'requires_positive_y': False,
+    'X_types': ['2darray'],
+    'poor_score': False,
+    'no_validation': False,
+    'multioutput': False,
+    "allow_nan": False,
+    'stateless': False,
+    'multilabel': False,
+    '_skip_test': False,
+    '_xfail_checks': False,
+    'multioutput_only': False,
+    'binary_only': False,
+    'requires_fit': True,
+    'preserves_dtype': [np.float64],
+    'requires_y': False,
+    'pairwise': False,
+}
+
+
+def _safe_tags(estimator, key=None, default=None):
+    """Safely get estimator tags for common checks.
+
+    :class:`~sklearn.BaseEstimator` provides the estimator tags machinery.
+    However, if a compatible estimator does not inherit from this base class,
+    we should default to the default tag.
+
+    Parameters
+    ----------
+    estimator : estimator object
+        The estimator from which to get the tag.
+
+    key : str, default=None
+        Tag name to get. By default (`None`), all tags are returned.
+
+    default : list of {str, dtype} or bool, default=None
+        When `esimator.get_tags()` is not implemented, default` allows to
+        define the default value of a tag if it is not present in
+        `_DEFAULT_TAGS` or to overwrite the value in `_DEFAULT_TAGS` if it the
```

Comment:
Do we really want to support third parties to implement their *own* tags?

Reply:
I would say yes.

Reply:
This is not needed at the moment IMO. Maybe we can leave this feature for later, in order to keep minimal here?